### PR TITLE
Fix nested drop downs + test scenarios for drop-down and vector widgets.

### LINKED ISSUE #8822: doubly nested dropdown options are not propagating to widgets
Table.filter should have a dropdown available for "Equals" inside filter parameter. When filling in the "Equals" parameter, brackets must be inserted.

Aggregate needs to have a "columns" dropdown for vector entries as well.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Part of #8518 
Closes #8822 

The drop downs were missing, because WidgetFunction had a higher score than them and took precedence. The fix was to adjust scores, so drop down is before both argument name and function. Generally, any widget expecting to "overlap" existing code (and possibly further widgets) should go before WidgetFunction.

### Important Notes

* Extended mockExpressionUpdate to accept a subexpression of a node. Also, I removed it from GraphDb - I want this function to have as little impact on app code as possible.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/e2e/widgets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    'Aggregate_Column',
+    '.',
+    'Group_By',
+    '"',
+    'column 2',
+    '"',
+  ])
+
+  // Switch aggregates
+  //TODO[ao] I have no idea how to emulate drag. Simple dragTo does not work (some element seem to capture event).
```
Comment (MichaelMauderer) [inline app/gui2/e2e/widgets.spec.ts:222]: Should we link this TODO to a proper issue?

File: app/gui2/e2e/collapsingAndEntering.spec.ts [hunk 2]
```diff
@@ -93,12 +94,16 @@ test('Collapsing nodes', async ({ page }) => {
 })
 
 async function expectInsideMain(page: Page) {
-  await expect(locate.graphNode(page)).toHaveCount(5)
+  await expect(locate.graphNode(page)).toHaveCount(9)
```
Comment (MichaelMauderer) [inline app/gui2/e2e/collapsingAndEntering.spec.ts:97]: Would it be possible to have this constant live with the source code where the initial nodes are defined?

**Review by MichaelMauderer — APPROVED** [review]

File: app/gui2/e2e/widgets.spec.ts (hunk 1 above)
Comment (farmaazon) [reply app/gui2/e2e/widgets.spec.ts:222]: I added points in the e2e test task.

File: app/gui2/e2e/collapsingAndEntering.spec.ts (hunk 2 above)
Comment (farmaazon) [reply app/gui2/e2e/collapsingAndEntering.spec.ts:97]: Well, again I hit the problem, that the source is defined in `mock/engine.ts` and test files does not clear access to it. I will skip straightening that up ATM

**Pull request merged**